### PR TITLE
Well format the raw response when query parameter "pretty" enabled

### PR DESCRIPTION
**Pull request by LantaoJin**
### Description
Well format the `raw` response when query parameter `pretty` enabled (`/_plugins/_sql?format=raw&pretty=true` or `/_plugins/_sql?format=raw&pretty`)

#### Example:
$>curl -k -H 'Content-Type: application/json' -XPOST 'http://localhost:9200/_plugins/_sql?format=raw' -d '{"query": "SELECT type, manufacturer, customer_last_name, day_of_week, total_quantity, currency, taxless_total_price FROM opensearch_dashboards_sample_data_ecommerce LIMIT 5"}'

returns
```
type|manufacturer|customer_last_name|day_of_week|total_quantity|currency|taxless_total_price
order|Elitelligence|Underwood|Monday|2|EUR|36.98
order|Champion Arts|Bailey|Sunday|2|EUR|53.98
order|Low Tide Media|Butler|Sunday|2|EUR|199.98
order|Primemaster|Chandler|Sunday|2|EUR|174.98
order|Elitelligence|Weber|Monday|2|EUR|80.98
```

$>curl -k -H 'Content-Type: application/json' -XPOST 'http://localhost:9200/_plugins/_sql?format=raw&pretty=true' -d '{"query": "SELECT type, manufacturer, customer_last_name, day_of_week, total_quantity, currency, taxless_total_price FROM opensearch_dashboards_sample_data_ecommerce LIMIT 5"}'

returns
```
type |manufacturer   |customer_last_name|day_of_week|total_quantity|currency|taxless_total_price
order|Elitelligence  |Underwood         |Monday     |2             |EUR     |36.98
order|Champion Arts  |Bailey            |Sunday     |2             |EUR     |53.98
order|Low Tide Media |Butler            |Sunday     |2             |EUR     |199.98
order|Primemaster    |Chandler          |Sunday     |2             |EUR     |174.98
order|Elitelligence  |Weber             |Monday     |2             |EUR     |80.98
```

### Issues Resolved
https://github.com/opensearch-project/sql/issues/2718
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseFormatter.java [hunk 1]
```diff
@@ -55,22 +58,54 @@ public String format(Throwable t) {
   static class FlatResult {
     private final QueryResult response;
     private final boolean sanitize;
+    private final boolean pretty;
+
+    private List<String> headers;
+    private List<List<String>> data;
+    private int[] maxWidths;
```
Comment (dai-chen) [inline protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseFormatter.java:65]: Could you refactor this class a little? I think this class was focus on sanitize only. After this, the class becomes more and more complex...

Comment (vamsi-amazon) [reply protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseFormatter.java:65]: Although this is existing code, I don't think having inner static classes inside is a good pattern. While refactoring can we separte it out and have individual classes for Sanitizer and Prettifier if possible[Up to you how you refactor].

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

File: protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseFormatter.java (hunk 1 above)
Comment (dai-chen) [reply protocol/src/main/java/org/opensearch/sql/protocol/response/format/FlatResponseFormatter.java]: Could you add Javadoc for new class? In future, I think we can refactor base, sanitizer, prettier by decorator as needed, e.g. sanitize and then pretty.

**Review by vamsi-amazon — APPROVED** [review]

**Pull request merged**